src/fromager/resolver.py

```diff
@@ -184,149 +183,179 @@ def get_project_from_pypi(
     """Return candidates created from the project name and extras."""
     found_candidates: set[str] = set()
     ignored_candidates: set[str] = set()
-    simple_index_url = sdist_server_url.rstrip("/") + "/" + project + "/"
-    logger.debug("%s: getting available versions from %s", project, simple_index_url)
+    logger.debug("%s: getting available versions from %s", project, sdist_server_url)
 
+    client = pypi_simple.PyPISimple(
+        endpoint=sdist_server_url,
+        session=session,
+        accept=pypi_simple.ACCEPT_JSON_PREFERRED,
+    )
     try:
-        response = session.get(simple_index_url)
-        response.raise_for_status()
-        data = response.content
+        package = client.get_project_page(project)
     except Exception as e:
         logger.debug(
-            "%s: failed to fetch package index from %s: %s",
-            project,
-            simple_index_url,
+            "failed to fetch package index from %s: %s",
+            sdist_server_url,
             e,
         )
         raise
 
-    doc = html5lib.parse(data, namespaceHTMLElements=False)
-    for i in doc.findall(".//a"):
-        candidate_url = urljoin(simple_index_url, i.attrib["href"])
-        py_req = i.attrib.get("data-requires-python")
-        # PEP 658: Check for metadata availability (PEP 714 data-core-metadata first)
```

Comment:
Dumb Question: I see PEP 658 related code removed but not added in the this MR. Does PEP 792 cover PEP 658?
I am a bit confused. Am I missing anything?

Reply:
PEP 658 support is much simpler with `pypi-simple`, because the library handles all special cases. It's now a one-liner in the `Candidate()` constructor:
```
metadata_url=dp.metadata_url if dp.has_metadata else None
```